### PR TITLE
ch(store.ts): Setup redux toolkit for global state management[#187364794]

### PR DESCRIPTION
**Pull request by martinemahirwe**
##Configured the Redux store using Redux Toolkit

1.Created slices folder
2.Provided the redux store into the react app
3.Created custom hooks for useSelector and useDispatch with predefined types to avoid assigning types repeatedly
 [Delivers #187364794]

### CODE REVIEW

**Review by tuyishimehope — APPROVED** [review]
LGTM

**Review by Aldot-02 — APPROVED** [review]
LGTM

File: src/main.tsx [hunk 1]
```diff
@@ -2,9 +2,13 @@ import React from 'react'
 import ReactDOM from 'react-dom/client'
 import App from './App.tsx'
 import './index.css'
+import {store} from './store.ts';
+import { Provider } from 'react-redux'
 
 ReactDOM.createRoot(document.getElementById('root')!).render(
   <React.StrictMode>
+    <Provider store = {store}>
```
Comment (SergeInGithub) [inline src/main.tsx]: You can put this provider in the App.tsx file instead

File: src/Hooks.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { useDispatch,useSelector } from "react-redux";
+import { AppDispatch,RootState } from './store';
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
+export const useAppSelector = useSelector.withTypes<RootState>();
```
Comment (SergeInGithub) [inline src/Hooks.ts]: Create a hooks folder then and index.ts file inside that will export hooks in that folder. In the hooks folder you can then create a file for this then export everything from the index file

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Some reviews below @martinemahirwe Also rebase your branch

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**